ext/reflection/tests/ReflectionFunction_getParameter_basic.phpt

```diff
@@ -0,0 +1,41 @@
+--TEST--
+ReflectionFunction::getParameter()
+--FILE--
+<?php
+function foo(string $bar) {}
+
+$function = new ReflectionFunction('sort');
+var_dump($function->getParameter('array'));
+var_dump($function->getParameter('flags'));
+var_dump($function->getParameter(0));
+var_dump($function->getParameter(1));
```

Comment:
Uhhhh not sure how I feel about this, I think the first parameter should be ``1``, as this is what we report in error messages.

Reply:
> Uhhhh not sure how I feel about this, I think the first parameter should be `1`, as this is what we report in error messages.

I've updated this @Girgias, but, now that I think about it, it may be the error message that needs correcting. 

Now, the following code:

```php
$function = new ReflectionFunction('sort');
var_dump($function->getParameter(1)->getPosition());
```

Outputs:

```
int(0)
```

So we're asking for parameter 1, which is at position 0, and the code refers to it as position.

Reply:
eeeeeeeeeerrrrrrrrrggggggggg yeah okay, it takes position from 0 because previously the only way to get it was by getting all the params as an array...

This might require an email to internals in the end to discuss semantics (which sucks...) another idea is to add a new method which returns the position starting from 1? Not sure what it would be named tho...

Reply:
> eeeeeeeeeerrrrrrrrrggggggggg yeah okay, it takes position from 0 because previously the only way to get it was by getting all the params as an array...
> 
> This might require an email to internals in the end to discuss semantics (which sucks...) another idea is to add a new method which returns the position starting from 1? Not sure what it would be named tho...

I'll reset it for now and put the new method idea on a post-it somewhere, I can only imagine how well that'll go down on the mailing list